components/api/src/main/java/com/hotels/styx/api/messages/StreamingHttpResponse.java

```diff
@@ -0,0 +1,448 @@
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        StreamingHttpResponse other = (StreamingHttpResponse) obj;
+        return Objects.equal(this.version, other.version)
+                && Objects.equal(this.status, other.status)
+                && Objects.equal(this.headers, other.headers)
+                && Objects.equal(this.cookies, other.cookies);
+    }
+
+    /**
+     * Builder.
+     */
+    public static final class Builder {
+        private HttpResponseStatus status = OK;
+        private HttpHeaders.Builder headers;
+        private HttpVersion version = HTTP_1_1;
+        private boolean validate = true;
+        private Observable<ByteBuf> body;
+        private final List<HttpCookie> cookies;
+
+        public Builder() {
+            this.headers = new HttpHeaders.Builder();
+            this.body = Observable.empty();
+            this.cookies = new ArrayList<>();
+        }
+
+        public Builder(HttpResponseStatus status) {
+            this();
+            this.status = status;
+        }
+
+        public Builder(StreamingHttpResponse response) {
+            this.status = response.status();
+            this.version = response.version();
+            this.headers = response.headers().newBuilder();
+            this.body = response.body();
+            this.cookies = new ArrayList<>(response.cookies());
+        }
+
+        public Builder(HttpResponse response, Observable<ByteBuf> decoded) {
+            this.status = response.status();
+            this.version = response.version();
+            this.headers = response.headers().newBuilder();
+            this.body = decoded;
+            this.cookies = new ArrayList<>(response.cookies());
+        }
+
+        /**
+         * Sets the response status.
+         *
+         * @param status response status
+         * @return {@code this}
+         */
+        public Builder status(HttpResponseStatus status) {
+            this.status = requireNonNull(status);
+            return this;
+        }
+
+        /**
+         * Sets the response body.
+         *
+         * @param content response body
+         * @return {@code this}
+         */
+        public Builder body(Observable<ByteBuf> content) {
+            this.body = content;
+            return this;
+        }
+
+        /**
+         * Sets the HTTP version.
+         *
+         * @param version HTTP version
+         * @return {@code this}
+         */
+        public Builder version(HttpVersion version) {
+            this.version = requireNonNull(version);
+            return this;
+        }
+
+        /**
+         * Sets the value of the {@code Content-Type} header.
+         * @param contentType content type
+         *
+         * @return {@code this}
+         */
+        public Builder contentType(MediaType contentType) {
```

Comment:
Guava type appears in Styx API?

This comment probably applies to existing Full message types too.